File: ios/FabricNativeTextInputManager.m
#import <React/RCTViewManager.h>
#import <React/RCTUIManager.h>
#import <React/RCTViewManager.h>
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(FabricNativeTextInputManager, RCTViewManager)

RCT_EXPORT_VIEW_PROPERTY(value, NSString)
RCT_EXPORT_VIEW_PROPERTY(placeholder, NSString)
RCT_EXPORT_VIEW_PROPERTY(secureTextEntry, BOOL)
RCT_EXPORT_VIEW_PROPERTY(onChangeText, RCTDirectEventBlock)

@end

File: ios/FabricNativeTextInputManager.swift
import UIKit
import React

@objc(FabricNativeTextInputManager)
class FabricNativeTextInputManager: RCTViewManager {
  
  override func view() -> UIView! {
    return FabricNativeTextInputView()
  }
  
  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
}
